datamegh/api/storage/s3/s3.py

```diff
@@ -0,0 +1,56 @@
+""" API for AWS S3. """
+
+from boto3.s3.transfer import S3Transfer
+from boto3.exceptions import S3UploadFailedError
+from botocore.exceptions import ClientError
+
+from datamegh.commons.aws import get_aws_client
+from datamegh.util.logging import get_logger
+
+logger = get_logger("storage:s3")
+
+
+def upload(file_path, bucket_name, s3_object_name):
+    """
+    Method to upload file to s3 bucket
+    :param file_path: file path
+    :param bucket_name: s3 bucket name
+    :param s3_object_name: name of file to be in s3 bucket
+    """
+    s3_client = get_aws_client("s3")
+    transfer = S3Transfer(s3_client)
+    # Upload the file to S3
+    try:
+        transfer.upload_file(file_path, bucket_name, s3_object_name)
+    except FileNotFoundError:
+        logger.error("File: {} was not found".format(file_path))
```

Comment:
You cannot leave the exceptions unhandled (supressed) - if anything fails it should cascade up to the top level. We should just let the user / caller decide how to handle the exceptions. We are just the framework - we won't do anything unless our logic involves there. 

Just raise the exception again so that it's propagated upwards.
```suggestion
    except FileNotFoundError as e:
        logger.error("File: {} was not found".format(file_path))
        raise e
```